frontend/src/components/Login.js

```diff
@@ -51,25 +50,42 @@ class Register extends Component {
       this.setState({ isLoggedIn: this.isLoggedIn() });
     } else {
       // TODO: Display message if login wasn't successful
+      this.handleAPIErrors(message);
     }
   };
 
+  handleAPIErrors(message) {
+    console.log(message);
+    this.setState({
+      errors: [
+        ...this.state.errors,
+        { message: message, key: this.state.errors.length }
+      ]
+    });
```

Comment:
Hmmm, interesting. What exactly would cause a miss? Since it should trigger a refresh as soon as the error is appended, and the calling function won't continue to execute until the API call returns (with the promise). I definitely could be missing something though!

Reply:
When `setState` is called, (following terminology may be incorrect) the update is queued until the next lifecycle update. If the current state has no errors, and 2 calls return and trigger an error, they each send out a `setState` call with `{errors: [<call1's error>]}` and with `{errors: [<call2's error>]}` respectively. When React processes these updates, it *replaces* `state.errors` with what it receives, so call1's error gets swallowed.

Reply:
Definitely not a likely thing though, so.... yknow. What I meant by using an update function is:
```javascript
this.setState((state) => {
  errors: [
    ...state.errors,
    { message: message, key: state.errors.length }
  ]
});
```

Reply:
Ahh, really interesting point. As I have it now however, wouldn't React process call1, and then call2 after? So they don't execute immediately because they're async but when they do run, are they guaranteed to run synchronously (i.e. can we guarantee that consecutive `setState` calls finish before starting the next?) or is this the key problem?